src/session-marks.ts

```
@@ -0,0 +1,184 @@
+export const normalizeMarks = (raw: unknown): SessionMarks => {
+  const marks = emptyMarks();
+  if (!raw || typeof raw !== 'object') return marks;
+  const obj = raw as Record<string, unknown>;
+  const pins = obj.pins;
+  if (pins && typeof pins === 'object' && !Array.isArray(pins)) {
+    for (const [id, info] of Object.entries(pins as Record<string, unknown>)) {
+      if (!id || !info || typeof info !== 'object') continue;
+      const p = info as Record<string, unknown>;
+      marks.pins[id] = {
+        pinnedAt:
+          typeof p.pinnedAt === 'string' ? p.pinnedAt : new Date(0).toISOString(),
```

ESLint

[error] Replace `·?·p.pinnedAt` with `⏎············?·p.pinnedAt⏎···········`

(prettier/prettier)